source/agora/consensus/protocol/Nominator.d

```diff
@@ -909,6 +898,19 @@ extern(D):
                 fail_reason, data.prettify);
             return ValidationLevel.kInvalidValue;
         }
+        const Block last_block = this.ledger.getLastBlock();
+        if (last_block.header.height + 1 == slot_idx) // Let's check last block is still one before this one
+        {
+            const validators = last_block.header.validators.count();
+            auto signed = last_block.header.validators.setCount;
+            if (last_block.header.height > 1 // Genesis block is not signed
+                && signed <= validators / 2) // Not more than half have signed previous block
+            {
+                log.info("Waiting for more than half to sign the last block, signed={}",
+                    last_block.header.validators.setCount);
+                return ValidationLevel.kMaybeValidValue; // this node is not ready to continue but will not block progress
+            }
+        }
```

Comment:
We probably need an else case here, but I am not sure if it is in the scope of this PR.

Reply:
I think in the `else case` we are ok to return `ValidationLevel.kFullyValidatedValue` as we can assume other nodes had more than half the signatures and we received the block during catchup.

Reply:
what if `slot_idx >> height + 1`? then we are validating for a state that we dont know